tests/integration_tests/datasources/test_lxd_discovery.py

```diff
@@ -112,7 +112,6 @@ def test_lxd_datasource_discovery(client: IntegrationInstance):
     # Jammy not longer provides nocloud-net seed files (LP: #1958460)
     if ImageSpecification.from_os_image().release in [
```

Comment:
Since bionic image will eventually roll and eventually drop metadata too 
https://bugs.launchpad.net/cloud-images/+bug/1988401

To avoid having to re-fix CI once bionic images are built, let's actually test the condition that causes LXD to be chosen over nocloud instead of checking the series name != bionic.

`lxc config metadata show instance.name | grep no-cloud`

Here's a diff that I think will work for us today and avoid CI breakage once bionic daily images drop the no-cloud metadata:
```diff
diff --git a/tests/integration_tests/datasources/test_lxd_discovery.py b/tests/integration_tests/datasources/test_lxd_discovery.py
index ddf4642d9..e6e2a08dd 100644
--- a/tests/integration_tests/datasources/test_lxd_discovery.py
+++ b/tests/integration_tests/datasources/test_lxd_discovery.py
@@ -8,7 +8,9 @@ from tests.integration_tests.instances import IntegrationInstance
 from tests.integration_tests.util import verify_clean_log
 
 
-def _customize_environment(client: IntegrationInstance):
+def _customize_environment(
+    client: IntegrationInstance, image_has_nocloud_metadata: bool
+):
     # Assert our platform can detect LXD during systemd generator timeframe.
     ds_id_log = client.execute("cat /run/cloud-init/ds-identify.log").stdout
     assert "check for 'LXD' returned found" in ds_id_log
@@ -36,7 +38,7 @@ def _customize_environment(client: IntegrationInstance):
         "datasource_list: [LXD, NoCloud]\n",
     )
     # This is also to ensure that NoCloud can be detected
-    if ImageSpecification.from_os_image().release == "jammy":
+    if not image_has_nocloud_metadata:
         # Add nocloud-net seed files because Jammy no longer delivers NoCloud
         # (LP: #1958460).
         client.execute("mkdir -p /var/lib/cloud/seed/nocloud-net")
@@ -54,7 +56,14 @@ def _customize_environment(client: IntegrationInstance):
 def test_lxd_datasource_discovery(client: IntegrationInstance):
     """Test that DataSourceLXD is detected instead of NoCloud."""
 
-    _customize_environment(client)
+    # Bionic++ no longer provide nocloud-net seed files (LP: #1988401)
+    lxd_image_metadata = subp(
+        ["lxc", "config", "metadata", "show", instance.name]
+    )
+    image_has_nocloud_metadata = bool(
+        "/var/lib/cloud/seed/nocloud-net" in lxd_image_metadata.stdout
+    )
+    _customize_environment(client, image_has_nocloud_metadata)
     result = client.execute("cloud-init status --wait --long")
     if not result.ok:
         raise AssertionError("cloud-init failed:\n%s", result.stderr)
@@ -109,10 +118,7 @@ def test_lxd_datasource_discovery(client: IntegrationInstance):
     )
     assert "#cloud-config\ninstance-id" in ds_cfg["meta-data"]
 
-    # Jammy not longer provides nocloud-net seed files (LP: #1958460)
-    if ImageSpecification.from_os_image().release in [
-        "bionic",
-    ]:
+    if image_has_nocloud_metadata:
         # Assert NoCloud seed files are still present in non-Jammy images
         # and that NoCloud seed files provide the same content as LXD socket.
         nocloud_metadata = yaml.safe_load(
diff --git a/tests/integration_tests/modules/test_combined.py b/tests/integration_tests/modules/test_combined.py
index b264f700d..2417d169b 100644
--- a/tests/integration_tests/modules/test_combined.py
+++ b/tests/integration_tests/modules/test_combined.py
@@ -76,6 +76,17 @@ timezone: US/Aleutian
 @pytest.mark.ci
 @pytest.mark.user_data(USER_DATA)
 class TestCombined:
+    def lxd_has_nocloud_metadata(
+        self, class_client: IntegrationInstance
+    ) -> bool:
+        # Bionic++ no longer provide nocloud-net seed files (LP: #1988401)
+        lxd_image_metadata = subp(
+            ["lxc", "config", "metadata", "show", client.name]
+        )
+        return bool(
+            "/var/lib/cloud/seed/nocloud-net" in lxd_image_metadata.stdout
+        )
+
     def test_final_message(self, class_client: IntegrationInstance):
         """Test that final_message module works as expected.
 
@@ -217,7 +228,7 @@ class TestCombined:
         parsed_datasource = json.loads(status_file)["v1"]["datasource"]
 
         if client.settings.PLATFORM in ["lxd_container", "lxd_vm"]:
-            if ImageSpecification.from_os_image().release == "bionic":
+            if self.lxd_has_nocloud_metadata(client):
                 datasource = "DataSourceNoCloud"
             else:
                 datasource = "DataSourceLXD"
@@ -291,7 +302,7 @@ class TestCombined:
         data = json.loads(instance_json_file)
         self._check_common_metadata(data)
         v1_data = data["v1"]
-        if ImageSpecification.from_os_image().release != "bionic":
+        if not self.lxd_has_nocloud_metadata(client):
             cloud_name = "lxd"
             subplatform = "LXD socket API v. 1.0 (/dev/lxd/sock)"
             # instance-id should be a UUID
@@ -327,7 +338,7 @@ class TestCombined:
         data = json.loads(instance_json_file)
         self._check_common_metadata(data)
         v1_data = data["v1"]
-        if ImageSpecification.from_os_image().release != "bionic":
+        if not self.lxd_has_nocloud_metadata(client):
             cloud_name = "lxd"
             subplatform = "LXD socket API v. 1.0 (/dev/lxd/sock)"
             # instance-id should be a UUID
```

Reply:
taking this to a separate PR